framework/src/node/forger/forger.ts

```diff
@@ -191,16 +198,52 @@ export class Forger {
 			);
 		}
 
-		const account = await this._chainModule.dataAccess.getAccountByAddress(forgerAddress);
-
-		if (forging) {
-			this._keypairs.set(forgerAddress, keypair);
-			this._logger.info(`Forging enabled on account: ${account.address.toString('hex')}`);
-		} else {
+		if (!forging) {
+			// Disable delegate by removing keypairs corresponding to address
 			this._keypairs.delete(forgerAddress);
-			this._logger.info(`Forging disabled on account: ${account.address.toString('hex')}`);
+			this._logger.info(`Forging disabled on account: ${forgerAddress.toString('hex')}`);
+			return {
+				address: forgerAddress,
+				forging,
+			};
 		}
 
+		const lastBlockHeader = this._chainModule.lastBlock.header;
+		const previouslyForgedMap = await getPreviouslyForgedMap(this._db);
+		const forgingInput = { height, maxHeightPreviouslyForged, maxHeightPrevoted };
+
+		if (!isSyncedWithNetwork(lastBlockHeader, forgingInput)) {
+			throw new Error('Failed to enable forging as the node is not synced to the network.');
+		}
+
+		if (!overwrite) {
+			// check if forger info exists
+			if (!previouslyForgedMap.has(forgerAddress)) {
```

Comment:
if delegate is forging for a first time, this will always fail. I think originally it was checking if the values are all zeros to see if it's first time or not?

Reply:
So with the current code the node operator has to call `updateForgingStatus` with `overwrite==True` and `height`, `maxHeightPreviouslyForged`, `maxHeightPrevoted` set to `0`.

Reply:
yes, i think it would be better if `overwrite = true` should only be used in the exception case. in the normal scenario, so it should never be specified.